app/controllers/notes_controller.rb

```diff
@@ -0,0 +1,26 @@
+class NotesController < ApplicationController
+
+  def index
+    @notes = Note.all
+    render json: @notes
+  end
+
+  def create
+    @note = Note.new(note_params)
+    if @note.save
+      params[:tags].split(",").map(&:strip).each do |tag|
+        @note.tags << Tag.find_or_initialize_by(name: tag)
+      end
+      render json: @note
+    else
+      render json: {errors: @note.errors.full_messages.map{|e| {error: e}}}, status: 400
```

Comment:
The test file expects the errors to be in this format:

```
{ errors: 
  [
    {error: "Title cannot be blank"},
    {error: "Body cannot be blank"}
  ]
}
```

So, we construct a hash by hand by looping over the `full_messages` of the errors and collecting a hash for each of them.

We're also setting a status of 400 here.